torch/csrc/jit/codegen/cuda/kernel_cache.h

```diff
@@ -13,21 +13,125 @@ namespace jit {
 namespace fuser {
 namespace cuda {
 
+// Go through a tensor, and grab it's sizes/strides potentially broadcasted
+struct ExtractSizeStride {
+  std::vector<int64_t> sizes;
+  std::vector<int64_t> strides;
+
+  // TODO: broadcasted_size should be handled in codegen directly instead of at
+  //       the integration limbo.
+  explicit ExtractSizeStride(
+      const at::Tensor& val,
+      c10::optional<at::IntArrayRef> broadcasted_size = c10::nullopt) {
+    if (broadcasted_size) {
+      // [Note - broadcast support in integration]
+      // PyTorch follows numpy broadcasting rule.
+      // (https://numpy.org/doc/stable/user/basics.broadcasting.html)
+      //
+      // So in case where the rank of two operators differ, we align them on
+      // the higher dimensions, hence the offset o_dim-b_dim to the index here.
+      int b_dim = (int)broadcasted_size->size();
```

Comment:
nit: functional style cast instead of c-style casts

`int(broadcasted_size->size())`